packages/icon-button/src/IconButton.tsx

```diff
@@ -82,37 +85,45 @@ const iconButtonSizes = {
   `,
 } as const;
 
-const iconButtonMode = {
-  [Mode.Light]: css`
-    &:hover {
-      color: ${uiColors.gray.dark2};
-
-      &:before {
-        background-color: ${uiColors.gray.light2};
+const getIconButtonModeStyle = (
+  isTouchDevice: boolean,
```

Comment:
I might be missing something, but I don't think this is being used anymore. We can just go back to the old structure now in this case.